frontend/app/components/session-timeout.tsx

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { type IIdleTimerProps, useIdleTimer } from 'react-idle-timer';
+
+import { Button } from './buttons';
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle } from '~/components/dialog';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('gcweb');
+
+export interface SessionTimeoutProps extends Pick<IIdleTimerProps, 'promptBeforeIdle'>, Pick<IIdleTimerProps, 'timeout'> {}
+
+const SessionTimeout = ({ promptBeforeIdle, timeout }: SessionTimeoutProps) => {
+  const { t } = useTranslation(i18nNamespaces);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [timeRemaining, setTimeRemaining] = useState('');
+
+  const handleOnIdle = () => {
+    window.location.href = '/auth/logout';
+  };
+
+  const handleOnIdleContinueSession = () => {
+    setModalOpen(false);
+    reset();
+  };
+
+  const { reset, getRemainingTime } = useIdleTimer({
+    onIdle: handleOnIdle,
+    onPrompt: () => setModalOpen(true),
+    promptBeforeIdle: promptBeforeIdle ?? 5 * 60 * 1000, //5 minutes
+    timeout: timeout ?? 20 * 60 * 1000, //20 minutes
+  });
+
+  const tick = useCallback(() => {
+    const minutes = Math.floor(getRemainingTime() / 60000);
+    const seconds = Math.floor((getRemainingTime() / 1000) % 60).toFixed(0);
+    setTimeRemaining(minutes + ':' + (parseInt(seconds) < 10 ? '0' : '') + seconds);
```

Comment:
```suggestion
    const minutes = Math.floor(getRemainingTime() / 60_000);

    const remainingSeconds = getRemainingTime() % 60_000;
    const seconds = Math.floor(remainingSeconds / 1_000);

    const formattedMinutes = minutes.toString().padStart(1, '0');
    const formattedSeconds = seconds.toString().padStart(2, '0');

    setTimeRemaining(`${formattedMinutes}:${formattedSeconds}`);
```

I feel like this implementation is slightly more readable (particularly the string template at the end).

But if you don't agree, just say so.

Reply:
updated